qiskit_ibm_runtime/qpy/__init__.py

```diff
@@ -0,0 +1,637 @@
+``'p'`` defines a :class:`~qiskit.circuit.Parameter` object  which is
+represented by a :ref:`qpy_param_struct` struct, ``e`` defines a
+:class:`~qiskit.circuit.ParameterExpression` object (that's not a
+:class:`~qiskit.circuit.Parameter`) which is represented by a :ref:`qpy_param_expr`
+struct (on QPY format :ref:`qpy_version_3` the format is tweak slightly see:
+:ref:`qpy_param_expr_v3`), ``'n'`` represents an object from numpy (either an
+``ndarray`` or a numpy type) which means the data is .npy format [#f2]_ data,
+and in QPY :ref:`qpy_version_3` ``'v'`` represents a
+:class:`~qiskit.circuit.ParameterVectorElement` which is represented by a
+:ref:`qpy_param_vector` struct.
+
+.. _qpy_param_struct:
+
+PARAMETER
+---------
+
+A PARAMETER represents a :class:`~qiskit.circuit.Parameter` object the data for
+a INSTRUCTION_PARAM. The contents of the PARAMETER are defined as:
+
+.. code-block:: c
+
+    struct {
+        uint16_t name_size;
+        char uuid[16];
+    }
+
+which is immediately followed by ``name_size`` utf8 bytes representing the
+parameter name.
+
+.. _qpy_param_expr:
+
+PARAMETER_EXPR
+--------------
+
+A PARAMETER_EXPR represents a :class:`~qiskit.circuit.ParameterExpression`
+object that the data for an INSTRUCTION_PARAM. The contents of a PARAMETER_EXPR
+are defined as:
+
+The PARAMETER_EXPR data starts with a header:
+
+.. code-block:: c
+
+    struct {
+        uint64_t map_elements;
+        uint64_t expr_size;
+    }
+
+Immediately following the header is ``expr_size`` bytes of utf8 data containing
+the expression string, which is the sympy srepr of the expression for the
+parameter expression. Follwing that is a symbol map which contains
+``map_elements`` elements with the format
+
+.. code-block:: c
+
+    struct {
+        char type;
+        uint64_t size;
+    }
+
+Which is followed immediately by ``PARAMETER`` object (both the struct and utf8
+name bytes) for the symbol map key. That is followed by ``size`` bytes for the
+data of the symbol. The data format is dependent on the value of ``type``. If
+``type`` is ``p`` then it represents a :class:`~qiskit.circuit.Parameter` and
+size will be 0, the value will just be the same as the key. If
+``type`` is ``f`` then it represents a double precision float. If ``type`` is
+``c`` it represents a double precision complex, which is represented by :ref:`qpy_complex`.
+Finally, if type is ``i`` it represents an integer which is an ``int64_t``.
+
+.. _qpy_complex:
+
+COMPLEX
+-------
+
+When representing a double precision complex value in QPY the following
+struct is used:
+
+
+.. code-block:: c
+
+    struct {
+        double real;
+        double imag;
+    }
+
+this matches the internal C representation of Python's complex type. [#f3]_
+
+
+.. [#f1] https://tools.ietf.org/html/rfc1700
+.. [#f2] https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html
+.. [#f3] https://docs.python.org/3/c-api/complex.html#c.Py_complex
```

Comment:
You don't need any of this, the protocol format specification is part of terra, you can just link to that in https://qiskit.org/documentation/

```suggestion
```